lib/eve.js

```diff
@@ -0,0 +1,63 @@
+import { useState, useEffect } from 'preact/hooks'
+
+// throttle a function call to avoid calling it too much
+export const throttle = (fn, delay, t, call = () => ((t = 0), fn())) => () =>
+  t || (t = setTimeout(call, delay))
+
+// create an observable or event
+export const eve = (...args) => {
+  const subs = new Set()
+  const sub = (fn) => (subs.add(fn), () => subs.delete(fn))
+  const next = () => new Promise(once)
+  const once = (fn) =>
+    subs.add(function $(value, prev) {
+      subs.delete($)
+      fn(value, prev)
+    })
+
+  // If no arguments are passed, this is an simple event
+  if (!args.length) {
+    const trigger = (value) => {
+      for (const fn of subs) fn(value)
+    }
+    return { next, once, on: sub, trigger }
+  }
+
+  // Otherwhise it's an observable with `set`, `get` and `map`
+  // additional methods
+  let [data] = args
```

Comment:
I take the first argument with destructuring, that will be our initial value and updated to the current value of the observale, this is where the data is kept